ircd/m_part.c

```diff
@@ -161,6 +161,9 @@ int m_part(struct Client* cptr, struct Client* sptr, int parc, char* parv[])
     if ((member->channel->mode.mode & MODE_NOCOLOR) && colors)
       flags |= CHFL_BANNED;
 
+    if (member->channel->mode.mode & MODE_NOPARTMSGS)
```

Comment:
At this point we know the client is allowed to send messages to the channel.  Why block their part messages but not their privmsgs?

Reply:
Because the purpose of the mode is to prevent part and quit messages being used for spam or for other abuse.  The user might still be able to speak with privmsg (subject to other modes), but the channel has decided they don't want part or quit messages being visible.

Reply:
Those "other modes" should have been checked by the call to member_can_send_to_channel() or the MODE_NOCOLOR (+c) check.  A spammer could just cause his bots to send PRIVMSGs instead of PART or QUIT spam, and make this mode ineffective.

Reply:
Well, we have channel mode +m to prevent PRIVMSGs.  The purpose of this one is not to prevent those.  It's an **additional** tool available to channels to combat abuse-- one that is different to what existing modes provide.  It can be used in combination with others, to help reduce the impact that abusers can have on channels.